app/feature/feature-travel-certificate/src/main/kotlin/com/hedvig/android/feature/travelcertificate/ui/choose/ChooseContractForCertificateViewModel.kt

```diff
@@ -0,0 +1,72 @@
+package com.hedvig.android.feature.travelcertificate.ui.choose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import com.hedvig.android.data.travelcertificate.ContractEligibleWithAddress
+import com.hedvig.android.data.travelcertificate.GetEligibleContractsWithAddressUseCase
+import com.hedvig.android.logger.LogPriority
+import com.hedvig.android.logger.logcat
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+
+internal class ChooseContractForCertificateViewModel(
+  getEligibleContractsWithAddressUseCase: GetEligibleContractsWithAddressUseCase,
+) : MoleculeViewModel<ChooseContractEvent, ChooseContractUiState>(
+    initialState = ChooseContractUiState.Loading,
+    presenter = ChooseContractPresenter(getEligibleContractsWithAddressUseCase),
+  )
+
+internal class ChooseContractPresenter(
+  private val getEligibleContractsWithAddressUseCase: GetEligibleContractsWithAddressUseCase,
+) : MoleculePresenter<ChooseContractEvent, ChooseContractUiState> {
+  @Composable
+  override fun MoleculePresenterScope<ChooseContractEvent>.present(
+    lastState: ChooseContractUiState,
+  ): ChooseContractUiState {
+    var dataLoadIteration by remember { mutableIntStateOf(0) }
+    var currentState by remember {
+      mutableStateOf<ChooseContractUiState>(ChooseContractUiState.Loading)
```

Comment:
```suggestion
      mutableStateOf<ChooseContractUiState>(lastState)
```
Here we can use what the state was before (if there was anything before) for the initial value